app/api/user/router.py

```diff
@@ -1,29 +1,32 @@
-from typing import Annotated
-import bcrypt
-from sqlalchemy.orm import Session
 from starlette import status
-from fastapi import APIRouter, Depends
+from fastapi import APIRouter, HTTPException, Path
 from app.api.user.schemas import UserRequest, UserResponse
-from app.core.utils import get_db
+from app.core.security import get_password_hash
 from app.models import User
+from app.core.utils import db_dependency
 
 router = APIRouter()
 
-db_dependency = Annotated[Session, Depends(get_db)]
-
 @router.get('/', status_code=status.HTTP_200_OK, response_model=list[UserResponse])
 async def get_all_users(db: db_dependency):
     return db.query(User).all()
 
+@router.get('/getByUsername/{username}', status_code=status.HTTP_200_OK, response_model=UserResponse)
```

Comment:
em geral se usa assim:
```suggestion
@router.get('/{username}', status_code=status.HTTP_200_OK, response_model=UserResponse)
```
e o melhor é buscar pelo id ao invés do username

Reply:
feito